lib/mumukit/core/class.rb

```diff
@@ -0,0 +1,20 @@
+class Class
+  class ContractBreachError < Exception
+  end
+
+  class AbstractClassInstantiationError < Exception
+  end
+
+  unless %w(RACK_ENV RAILS_ENV).any? { |it| ENV[it] == 'production' }
+    alias_method :__new__, :new
+
+    def new(*args, &block)
+      validate_upholds_contract!
+      __new__(*args, &block)
+    end
+  end
+
+  def validate_upholds_contract!
```

Comment:
I find the term `uphold` weird. I think _meeting a contract_ or _honouring a contract_ are more common terms, aren't them?

Reply:
hmmm, how about `complies_with`?